network/server/tchannelthrift/convert/convert.go

```diff
@@ -175,6 +187,163 @@ func ToRPCError(err error) *rpc.Error {
 	return tterrors.NewInternalError(err)
 }
 
+// FromRPCFetchTaggedRequest converts the rpc request type for FetchTaggedRequest into corresponding Go API types.
+func FromRPCFetchTaggedRequest(
+	req *rpc.FetchTaggedRequest,
+) (ident.ID, index.Query, index.QueryOptions, bool, error) {
+	start, rangeStartErr := ToTime(req.RangeStart, fetchTaggedTimeType)
+	if rangeStartErr != nil {
+		return nil, index.Query{}, index.QueryOptions{}, false, rangeStartErr
+	}
+
+	end, rangeEndErr := ToTime(req.RangeEnd, fetchTaggedTimeType)
+	if rangeEndErr != nil {
+		return nil, index.Query{}, index.QueryOptions{}, false, rangeEndErr
+	}
+
+	opts := index.QueryOptions{
+		StartInclusive: start,
+		EndExclusive:   end,
+	}
+	if l := req.Limit; l != nil {
+		opts.Limit = int(*l)
+	}
+
+	q, err := fromRPCQuery(req.Query)
+	if err != nil {
+		return nil, index.Query{}, index.QueryOptions{}, false, err
+	}
+
+	return ident.StringID(req.NameSpace), index.Query{q}, opts, req.FetchData, nil
```

Comment:
Hm, do you want to pass in a ident.Pool perhaps and if not nil then use it in place, etc?  Or we'll do this later?